grails-app/controllers/org/bbop/apollo/AvailableStatusController.groovy

```diff
@@ -110,4 +112,177 @@ class AvailableStatusController {
             '*'{ render status: NOT_FOUND }
         }
     }
+
+    @RestApiMethod(description = "Create status", path = "/availableStatus/createStatus", verb = RestApiVerb.POST)
+    @RestApiParams(params = [
+            @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "name", type = "string", paramType = RestApiParamType.QUERY, description = "Status name to add")
+    ]
+    )
+    @Transactional
+    def createStatus() {
+        JSONObject statusJson = permissionService.handleInput(request, params)
+        try {
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(statusJson))) {
+                if (statusJson.get("name") == "") {
+                    throw new Exception('empty fields detected')
+                }
+
+                log.debug "Adding ${statusJson.name}"
+                AvailableStatus status = new AvailableStatus(
+                        label: statusJson.name
+                ).save(flush: true)
+
+                render status as JSON
+            } else {
+                def error = [error: 'not authorized to add AvailableStatus']
+                render error as JSON
+                log.error(error.error)
+            }
+        } catch (e) {
+            def error = [error: 'problem saving AvailableStatus: ' + e]
+            render error as JSON
+            e.printStackTrace()
+            log.error(error.error)
+        }
+    }
+
+    @RestApiMethod(description = "Update status", path = "/availableStatus/updateStatus", verb = RestApiVerb.POST)
+    @RestApiParams(params = [
+            @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "id", type = "long", paramType = RestApiParamType.QUERY, description = "Status ID to update (or specify the old_name)")
+            , @RestApiParam(name = "old_name", type = "string", paramType = RestApiParamType.QUERY, description = "Status name to update")
+            , @RestApiParam(name = "new_name", type = "string", paramType = RestApiParamType.QUERY, description = "Status name to change to (the only editable option)")
+    ]
+    )
+    @Transactional
+    def updateStatus() {
+        try {
+            JSONObject statusJson = permissionService.handleInput(request, params)
+            log.debug "Updating status ${statusJson}"
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(statusJson))) {
+
+                log.debug "status ID: ${statusJson.id}"
+                AvailableStatus status = AvailableStatus.findById(statusJson.id)
+                if (!status) {
+                    status = AvailableStatus.findByLabel(statusJson.old_name)
+                }
```

Comment:
in groovy you can write: 

```AvailableStatus status = AvailableStatus.findById(statusJson.id) ?: status = AvailableStatus.findByValue(statusJson.old_name)```

Also, remember to use value instead of label.  If you need both, then you'll need to add the label property to AvailableStatus.